### PR TITLE
CI: Add new UMA strategy test from stsp

### PR DESCRIPTION
**Pull request by andrewbird**
See #1857

### CODE REVIEW

File: test/func_memory_uma.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+$_hdimage = "dXXXXs/c:hdtype1 +1"
+$_floppy_a = ""
+"""
+
+
+def memory_uma_strategy(self):
+
+# Note: Not sure if I need this
+    if 'FDPP' in self.version:
+        self.mkfile("userhook.sys", """DOS=HIGH\n""", newline="\r\n")
```
Comment (stsp) [inline test/func_memory_uma.py]: What does this do?

Comment (stsp) [reply test/func_memory_uma.py]: I think its not needed, not testing HMA
here.

Comment (andrewbird) [reply test/func_memory_uma.py]: Okay, cool I'll remove it.

BTW, this test passes on DR-DOS 7.01, MS-DOS 6.22, MS-DOS 7.0, MS-DOS 7.1 and FDPP, but it **fails** on FreeDOS 1.2. Is that expected?

Comment (stsp) [reply test/func_memory_uma.py]: It seems to work if not use `-dumb`.
UMB/HMA support in freedos is very,
very rudimentary as we know from
all the rewrites that were done.
So while not expected, its also not
very surprising.

Comment (stsp) [reply test/func_memory_uma.py]: Shouldn't we move to freedos 1.3?
I think it was quite active (esp in
back-porting fdpp patches), so maybe
more tests will work?

Comment (andrewbird) [reply test/func_memory_uma.py]: > Shouldn't we move to freedos 1.3?

Yes it's on my personal list of things to do.

**Comment by stsp** [comment]
How are you running the tests?
With some extra call to command.com?
I wonder why it seems like the
"default loadhigh" strategy doesn't
affect tests.

**Comment by andrewbird** [comment]
> How are you running the tests?

Exactly the same as if you typed 'testit' at the DOS prompt. Since I use [pexpect](https://pypi.org/project/pexpect/) dosemu should not be aware that it's not interacting with a human at the terminal.

**Comment by stsp** [comment]
> Exactly the same as if you typed 'testit' at the DOS prompt.

This means you are not using
-E/-K, but some other machinery
of yours?

**Comment by andrewbird** [comment]
> This means you are not using
-E/-K, but some other machinery
of yours?

Yes that's correct, except for a very few tests that actually test that, plus pcmos and libi86.

**Comment by stsp** [comment]
That's quite sad actually, because
the switch to "lh by default" crashed
only the tests you mentioned. The
rest were unaffected, and now appears
they were ignoring that change.

**Comment by stsp** [comment]
Thanks.

**Pull request merged**